tools/scoring/score/dimensions/dimension.py

```diff
@@ -11,71 +11,103 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Core functionality parent class"""
+"""Core component base class"""
 
-from dataclasses import dataclass
+from typing import Any, Dict, List, Tuple
 
 
-@dataclass
 class Dimension:
   """
     Container for floating-point results which
     quantify success in each dimension
 
     Attributes:
-      correct_virtual: Number of attempts achieved within virtual devices
-      correct_reporting: Number of attempts achieved within reporting devices
-      correct_ceiling_virtual: Number of attempts possible to achieve
+      translations: Proposed and solution translations
+        for all matched reporting entities. Assigned via argument
+      deserialized_files: Parsed configuration files.  Assigned via argument
+
+      correct_virtual: Number of successful attempts within virtual devices
+      correct_reporting: Number of successful attempts within reporting devices
+      correct_ceiling_virtual: Number of attempts possible
         within virtual devices
-      correct_ceiling_reporting: Number of attempts possible to achieve
-        within reporting devices
-      incorrect_virtual: Number of attempts not achieved within virtual devices
-      incorrect_reporting: Number of attempts not achieved
+      correct_ceiling_reporting: Number of attempts possible
         within reporting devices
+      incorrect_virtual: Number of failed attempts within virtual devices
+      incorrect_reporting: Number of failed attempts within reporting devices
 
     Properties:
       result_composite: Calculated result for all devices
       result_virtual: Calculated result for virtual devices
       result_reporting: Calculated result for reporting devices
   """
-  correct_virtual: int
-  correct_reporting: int
-  correct_ceiling_virtual: int
-  correct_ceiling_reporting: int
-  incorrect_virtual: int
-  incorrect_reporting: int
+  def __init__(self,
+               *,
+               translations: Dict[str, List[Tuple[str, Any]]] = None,
+               deserialized_files: Any = None):
+    self.translations = translations
+    self.deserialized_files = deserialized_files
+    # self.type: Literal['simple', 'complex'] = None
+
+    self.correct_virtual: int = None
+    self.correct_reporting: int = None
+    self.correct_ceiling_virtual: int = None
+    self.correct_ceiling_reporting: int = None
+    self.incorrect_virtual: int = None
+    self.incorrect_reporting: int = None
 
-  def correct(self) -> int:
-    """ Number of attempts achieved within all devices """
-    return self.correct_virtual + self.correct_reporting
+    if not translations and not deserialized_files:
+      raise Exception(
+          '`translations` xor `deserialized_files` argument is required')
```

Comment:
why do we need both? to look up virtual entities?

Reply:
Correct; the simple dimensions receive `translations` and the complex dimensions receive `deserialized_files`. (There's likely to be some repetition in the virtual entity matching algorithm(s) which will have to be combined down the line.)

Reply:
can you please leave a comment why you need both